app/static/js/bonnerManagement.js

```diff
@@ -69,7 +72,30 @@ function addRequirement() {
     addRequirementsRowHandlers()
     newRow.find("input").focus()
 }
-
+function addCohort(){
+    let years = $('#v-pills-tab .nav-link').map((i, element) => {return Number($(element).data('year'))}).get();
+    let newCohortYear = Math.max(...years) + 1;
+    // Remove class 'active' from the currently active tab
+    $('#v-pills-tab .active').removeClass('active');
+    $('#v-pills-tabContent .tab-pane').removeClass('show active')
+    $('#v-pills-tab #addCohort').after(`<button class="nav-link active" id="v-pills-${newCohortYear}-tab" data-bs-toggle="pill" data-bs-target="#v-pills-${newCohortYear}" type="button" role="tab" data-year="${newCohortYear}" aria-controls="v-pills-${newCohortYear}" aria-selected="{{aria}}">${newCohortYear} - ${newCohortYear + 1}</button>`)
+    $('#v-pills-tabContent').prepend(`
+    <div class="tab-pane fade show active" id="v-pills-${newCohortYear}" role="tabpanel" aria-labelledby="v-pills-${newCohortYear}-tab">
+        <div>
+            <div class="input-group mb-3">
+                <input type="search" id="search-${newCohortYear}" name="search-${newCohortYear}" class="form-control" data-year="${newCohortYear}" placeholder="Add Student" autocomplete="off" style="width:50%" />
+                <span class="input-group-text me-1"><span class="bi bi-search"></span></span>
+            </div>
+            <table class="w-100 table table-striped">
+                <tr><td>No students added.</td></tr>
+            </table>
+        </div>
+    </div>`)
+    $(`#search-${newCohortYear}`).on("input", function(){
```

Comment:
I could definitely be wrong but I believe this code might be unnecessary. I think this is duplicating functionality already being done on line 22.

Reply:
Good eyes on that, we did copy that logic from line 22 but we still need it. Since line 22 runs on page load and we're adding in new elements after page load, we need to rerun the logic that gives that input its searching capabilities. That said, we are copying code which is bad practice! I've abstracted this logic into a function called `addSearchCapabilities(inputElement)` which should solve the readability issues!